cpp/src/utils/check_status.h

```diff
@@ -2,7 +2,7 @@
 #define CHECK_GDAL(action)                                                  \
 {                                                                           \
     int check = action;                                                     \
```

Comment:
if the only possiable 'action' type is OGRERR, we may define CHECK_GDAL as follows for better clarity.
#define CHECK_GDAL(OGRERR)\
{\
    if(!!OGRERR){\
        std::string err_msg = "gdal error code = " + std::to_string(int(OGRERR)); \
        throw std::runtime_error(err_msg); \
}

Reply:
OK, I have rewrite `CHECK_GDAL` just as you suggest